opencsp/common/lib/cv/test/test_CacheableImage.py

```diff
@@ -0,0 +1,307 @@
+                np.testing.assert_array_equal(self.example_array, uncached_array)
+
+                # cache and delete the cache file
+                # loading from the cache should fail
+                cacheable.cache(default_cache_file)
+                if should_create_cache_file is not None:
+                    self.assertTrue(ft.file_exists(should_create_cache_file), msg=err_msg)
+                    ft.delete_file(should_create_cache_file)
+                    with self.assertRaises(Exception, msg=err_msg):
+                        cacheable.nparray
+                else:
+                    self.assertFalse(ft.file_exists(default_cache_file), msg=err_msg)
+                    self.assertFalse(ft.file_exists(noexist_cache_file), msg=err_msg)
+
+            except Exception:
+                lt.error(err_msg)
+                raise
+
+    def test_lru(self):
+        """Verifies that the Least Recently Used functionality works as expected"""
+
+        # create three cacheable images
+        # new LRU: 1, 2, 3
+        c1 = CacheableImage(self.example_array)
+        self.assertEqual(c1, CacheableImage.lru(False))
+        c2 = CacheableImage(self.example_array)
+        self.assertEqual(c1, CacheableImage.lru(False))
+        c3 = CacheableImage(self.example_array)
+        self.assertEqual(c1, CacheableImage.lru(False))
+
+        # get the value from the 1st image, then the 2nd image
+        # LRU: 2, 3, 1
+        c1.nparray
+        self.assertEqual(c2, CacheableImage.lru(False))
+        # LRU: 3, 1, 2
+        c2.nparray
+        self.assertEqual(c3, CacheableImage.lru(False))
+
+        # cache the 1st image, check that this doesn't change the lru
+        # LRU: 3, 2, 1
+        c1.cache(ft.join(self.out_dir, f"{self.test_name}_c1.npy"))
+        self.assertEqual(c3, CacheableImage.lru(False))
+
+        # get the value of the 3rd image, check that the 2nd is now the LRU
+        # LRU: 2, 1, 3
+        c3.nparray
+        self.assertEqual(c2, CacheableImage.lru(False))
+
+        # deregister the 2nd, then 1st, then 3rd
+        self.assertEqual(c2, CacheableImage.lru(True))
+        # LRU: 1, 3
+        self.assertEqual(c1, CacheableImage.lru(True))
+        # LRU: 3
+        self.assertEqual(c3, CacheableImage.lru(True))
+        # LRU:
+        self.assertEqual(None, CacheableImage.lru(True))
+
+        # get the value for and deregister the 1st, then 2nd, then 3rd
+        # LRU: 1
+        c1.nparray
+        self.assertEqual(c1, CacheableImage.lru(True))
+        # LRU: 2
+        c2.nparray
+        self.assertEqual(c2, CacheableImage.lru(True))
+        # LRU: 3
+        c3.nparray
+        self.assertEqual(c3, CacheableImage.lru(True))
+
+        # check that there are no more registered cacheable images
+        self.assertEqual(None, CacheableImage.lru(False))
+
+    def test_save_image(self):
+        """Verify that after saving the image, cacheing no longer creates a cache file"""
```

Comment:
Should this (and can it) check whether the in-memory np array & image are 0 bytes?